dashboard/src/reducers/userProfileReducer.js

```diff
@@ -0,0 +1,47 @@
+import {
+  GET_USER_DETAILS,
+  UPDATE_USER_DETAILS,
+  RESET_DATA,
+  SET_USER_DETAILS,
+} from "../actions/types";
+
+const initialState = {
+  userDetails: {},
+  userDetails_copy: {},
+  updatedUserDetails: {},
+  isUserDetailsUpdated: false,
+};
+
+const UserProfileReducer = (state = initialState, action = {}) => {
+  const { type, payload } = action;
+  switch (type) {
+    case GET_USER_DETAILS:
+      return {
+        ...state,
+        userDetails: Object.assign({}, payload),
+        userDetails_copy: Object.assign({}, payload),
```

Comment:
What is the purpose of `userDetails_copy`?  (When would it ever be different from `userDetails`?  Or, is it intended for (ab)use by the caller?)